src/main/java/org/apache/maven/plugins/assembly/mojos/AbstractAssemblyMojo.java

```diff
@@ -475,19 +475,20 @@ public void execute() throws MojoExecutionException, MojoFailureException {
         // TODO: include dependencies marked for distribution under certain formats
         // TODO: how, might we plug this into an installer, such as NSIS?
 
-        MavenArchiver mavenArchiver = new MavenArchiver();
-        Date outputDate = mavenArchiver.parseOutputTimestamp(outputTimestamp);
+        FileTime outputDate = MavenArchiver.parseBuildOutputTimestamp(outputTimestamp)
+                .map(FileTime::from)
+                .orElse(null);
```

Comment:
I find this new code so much more complicated...

Reply:
Harder to read?

Reply:
Methods with `Date` as argument are deprecated so I need a `FileTime`

Is there any other way to achieve it - or like code should look to be a less complicated?

Reply:
oh sadly not, it's the API that I don't understand why it has been done "as an improvement"

Reply:
plexus-archiver use FileTime after https://github.com/codehaus-plexus/plexus-archiver/issues/198

Reply:
@hboutemy , I don't think that you refer to the `FileTime`, but rather to the ultracompact lambda expressions, no?